sdk/python/kfp/compiler/compiler_utils.py

```diff
@@ -564,6 +570,11 @@ def _get_uncommon_ancestors(
     Returns:
         A tuple which are lists of uncommon ancestors for each task.
     """
+
+    if getattr(task1, 'is_task_group_dependent', False) or getattr(
+            task2, 'is_task_group_dependent', False):
+        return ([task1.name], [task2.name])
```

Comment:
This logic was added since exit task which has the exit handler as an upstream was[ removed from all groups and cannot have any dependent tasks](https://github.com/kubeflow/pipelines/blob/master/sdk/python/kfp/components/tasks_group.py#:~:text=if%20exit_task.,(exit_task)) in the default pipeline tasks. I have modified however to only check that the downstream is an exit task

Reply:
Thanks, Josh, for showing me offline that the root group should not be included included in the upstream groups nor the downstream groups.